chain/chain-primitives/src/error.rs

```diff
@@ -214,6 +214,16 @@ pub enum Error {
     Other(String),
 }
 
+impl std::convert::From<Error> for io::Error {
+    fn from(err: Error) -> io::Error {
+        match err {
+            Error::IOErr(err) => err,
+            Error::Other(msg) => io::Error::new(io::ErrorKind::Other, msg),
+            err => io::Error::new(io::ErrorKind::Other, err),
+        }
+    }
+}
+
```

Comment:
This feels slippery-sloppery to me, I don't think this impl makes sense on it's own, only in the context of us adding *some* kind of error reporting to migrations. 


I think what migrations *should* do is to use `anyhow`. So, as a minimal fix, I suggest adding `anyhow` to Cargo.toml here. This I'd say would be better (as this impl infects everything where we use `chain::Error`).

That wouldn't be perfect still -- `anyhow` at such a low-level code  as store feel wrong. I think this points at a bigger problem -- migrations shouldn't be part of the `store` crate. `store` ideally shouldn't know about chain types at all. So perhaps migrations should be moved to `nearcore` crate, which already depends on anyhow-style error propagation? We already do have some migrations there, WDYT about moving the rest there?

Reply:
Switched to anyhow

Reply:
Yeah, great point about `StoreOpener`. `anyhow` seems the right approach here. If we want to be pedantic, I think we could add 

```rust
#[thiserror::Error]
enum OpenStoreError {
    MigrationError(MigrationError)
    ...
}
```

where `MigrationError` is something which can type-erase errors from migrations passed in, but that's too much design for little real benefit.